test/800_plugin_cross_hosts_2_test.sh

```diff
@@ -15,7 +15,20 @@ weave_on $HOST2 launch $HOST1
 start_container_local_plugin $HOST1 --name=c1 --hostname=$C1_NAME --dns=$DNS_IP
 start_container_local_plugin $HOST2 --name=c2 --hostname=$C2_NAME --dns=$DNS_IP
 
+c1ip=$(container_ip $HOST1 c1)
+
 assert_raises "exec_on $HOST1 c1 $PING $C2_NAME"
 assert_raises "exec_on $HOST2 c2 $PING $C1_NAME"
 
+# Restart weave to check details are recovered
+weave_on $HOST1 stop-router
+weave_on $HOST1 launch-router
+
+start_container $HOST1 --name=c3
+
+assert "container_ip $HOST1 c1" "$c1ip"
+assert_raises "container_ip $HOST1 c3 | grep -v $c1ip"
+
+assert "weave_on $HOST1 dns-lookup $C1_NAME" "$c1ip"
+
```

Comment:
This should all go into a separate test suite - it has nothing to do with cross-host testing and does not require 2 hosts.